src/WebJobs.Script/Description/DotNet/Compilation/DotNetCompilationServiceFactory.cs

```diff
@@ -53,19 +53,19 @@ internal static void SetOptimizationLevel(OptimizationLevel? level)
             _optimizationLevel = level;
         }
 
-        public ICompilationService<IDotNetCompilation> CreateService(ScriptType scriptType, IFunctionMetadataResolver metadata)
+        public ICompilationService<IDotNetCompilation> CreateService(string language, IFunctionMetadataResolver metadata)
         {
-            switch (scriptType)
+            switch (language)
             {
-                case ScriptType.CSharp:
+                case "CSharp":
                     return new CSharpCompilationService(metadata, OptimizationLevel);
-                case ScriptType.FSharp:
+                case "FSharp":
                     return new FSharpCompilationService(metadata, OptimizationLevel, _loggerFactory);
-                case ScriptType.DotNetAssembly:
+                case "DotNetAssembly":
                     return new RawAssemblyCompilationService();
                 default:
                     throw new NotSupportedException(string.Format(CultureInfo.InvariantCulture,
-                        "The script type {0} is not supported by the {1}", scriptType, typeof(DotNetCompilationServiceFactory).Name));
+                        "The script type {0} is not supported by the {1}", language, typeof(DotNetCompilationServiceFactory).Name));
```

Comment:
I wonder if this text shouldn't be updated. I was thinking "The language {0} is not supported..." but "DotNetAssembly" isn't really a language. Hmmm.

Reply:
Perhaps file type?